cloudify_rest_client/inter_deployment_dependencies.py

```diff
@@ -0,0 +1,153 @@
+        self.api = api
+        self._uri_prefix = 'inter-deployment-dependencies'
+        self._wrapper_cls = InterDeploymentDependency
+
+    def _wrap_list(self, response):
+        return ListResponse(
+            [self._wrapper_cls(item) for item in response['items']],
+            response['metadata']
+        )
+
+    def create(self, dependency_creator, source_deployment, target_deployment):
+        """Creates an inter-deployment dependency.
+
+        :param dependency_creator: a string representing the entity that
+         is responsible for this dependency (e.g. an intrinsic function
+         blueprint path, 'node_instances.some_node_instance', etc.).
+        :param source_deployment: source deployment that depends on the target
+         deployment.
+        :param target_deployment: the deployment that the source deployment
+         depends on.
+        :return: an InterDeploymentDependency object.
+        """
+        data = {
+            'dependency_creator': dependency_creator,
+            'source_deployment': source_deployment,
+            'target_deployment': target_deployment
+        }
+        response = self.api.put(
+            '/{self._uri_prefix}'.format(self=self), data=data)
+        return self._wrapper_cls(response)
+
+    def delete(self,
+               dependency_creator,
+               source_deployment,
+               target_deployment,
+               doesnt_exist_ok=False):
+        """Deletes an inter-deployment dependency.
+
+        :param dependency_creator: a string representing the entity that
+         is responsible for this dependency (e.g. an intrinsic function
+         blueprint path, 'node_instances.some_node_instance', etc.).
+        :param source_deployment: source deployment that depends on the target
+         deployment.
+        :param target_deployment: the deployment that the source deployment
+         depends on.
+        :param doesnt_exist_ok: if it's False then no error will be raised if
+         the dependency doesn't exist. Defaults to False.
+        :return: an InterDeploymentDependency object.
+        """
+        data = {
+            'dependency_creator': dependency_creator,
+            'source_deployment': source_deployment,
+            'target_deployment': target_deployment,
+            'doesnt_exist_ok': doesnt_exist_ok
+        }
+        response = self.api.delete(
+            '/{self._uri_prefix}'.format(self=self), data=data)
+        return self._wrapper_cls(response)
+
+    def get(self,
+            dependency_creator,
+            source_deployment,
+            target_deployment,
+            _include=None,
+            **kwargs):
+        """
+        Gets an inter-deployment dependency by its id.
+
+        :param dependency_id: InterDeploymentDependency's id to get.
+        :param _include: List of fields to include in response.
+        :return: The inter-deployment dependency details.
+        """
+        assert dependency_creator
```

Comment:
plz no additional `assert`s, i do realize we have some of them, but it's pretty bad, because as you know asserts are entirely removed in "optimized mode" (`python -O`), but we do need those checks

insetad, please do something like
```python
if not (dependency_creator and source_deployment and target_deployment):
    raise ValueError('blabla')
```
feel free to write the `and` logic differently (3 separate `if`s are fine), and perhaps use CloudifyClientError, but a ValueError (or even RuntimeError) are better than asserts